src/Form/Project/ProcesVerbalSubType.php

```diff
@@ -11,22 +11,49 @@
 
 namespace App\Form\Project;
 
+use App\Entity\Project\Etude;
+use App\Entity\Project\Phase;
 use App\Entity\Project\ProcesVerbal;
-use Symfony\Component\Form\Extension\Core\Type\IntegerType;
+use App\Repository\Project\PhaseRepository;
+use Symfony\Bridge\Doctrine\Form\Type\EntityType;
 use Symfony\Component\Form\FormBuilderInterface;
 use Symfony\Component\OptionsResolver\OptionsResolver;
 
 class ProcesVerbalSubType extends DocTypeType
 {
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
-        $phaseNum = $options['phases'];
-        if ('pvi' == $options['type']) {
-            $builder->add(
-                'phaseID',
-                IntegerType::class,
-                ['label' => 'Phases concernées', 'required' => false, 'attr' => ['min' => '1', 'max' => $phaseNum]]
-            );
+        if (!isset($options['etude']) || !($options['etude'] instanceof Etude)) {
+            throw new \LogicException('A ProcesVerbalSubType can\'t be build without associated Etude object.');
+        }
+        $this->etude = $options['etude'];
```

Comment:
Ajout dynamique d'attributs à l'objet, c'est une mauvaise pratique. 

Et une variable locale devrait suffire. Éventuellement en utilisant [la propriété use ](https://stackoverflow.com/questions/1065188/in-php-what-is-a-closure-and-why-does-it-use-the-use-identifier )pour les closures.